### PR TITLE
translation of Infinispan

### LINKED ISSUE #32: 【vertx-infinispan】的翻译和校对
需要翻译总文件数：1
需要翻译总行数：452
详情如下：
    docs\translation\vertx-infinispan\java\index.adoc行数：452

### PR DESCRIPTION
**Pull request by Leibnizhu**
fix #32

### CODE REVIEW

File: docs/translation/vertx-infinispan/java/index.adoc [hunk 1]
```diff
@@ -13,30 +13,31 @@ This implementation is packaged inside:
...
+== 使用集群管理器
 
-If you are using Vert.x from the command line, the jar corresponding to this cluster manager (it will be named `vertx-infinispan-4.0.0.jar`
-should be in the `lib` directory of the Vert.x installation.
+如果通过命令行来使用 Vert.x，对应集群管理器的 `jar` 包（名为 `vertx-infinispan-4.0.0.jar` ）
+应该在 Vert.x 中安装路径的 `lib` 目录中。
 
-If you want clustering with this cluster manager in your Vert.x Maven or Gradle project then just add a dependency to
-the artifact: `io.vertx:vertx-infinispan:4.0.0` in your project.
+如果在 Maven 或者 Gradle 工程中使用 Vert.x， 只需要在工程依赖中加上依赖：
```
Comment (chengenzhao) [inline docs/translation/vertx-infinispan/java/index.adoc]:
```suggestion
若您希望在集群中使用此集群管理器，只需要在您的Vert.x Maven 或 Gradle 工程中添加依赖：
```

File: docs/translation/vertx-infinispan/java/index.adoc [hunk 2]
```diff
@@ -48,60 +49,62 @@ Vertx.clusteredVertx(options, res -> {
...
+前者配置数据网格（data grid），后者配置组管理和集群成员发现。
 
-You can place one or both of them on your classpath.
-If you want to embed your custom file in a fat jar, it must be located at the root of the fat jar.
-If it's an external file, the **directory** containing the file must be added to the classpath. For
-example, if you are using the _launcher_ class from Vert.x, the classpath enhancement can be done as follows:
+您可以在 classpath 中替换这些配置文件（其中一个或两个）。
+如果想在 fat jar 中内嵌自己的配置文件，这些文件必须在 fat jar 的根目录中。
+如果这些配置文件是外部文件，则必须将其所在的 **目录** 添加至 classpath 中。
+举个例子，如果使用 Vert.x 的 _launcher_ 启动应用，则 classpath 应该设置为：
```
Comment (chengenzhao) [inline docs/translation/vertx-infinispan/java/index.adoc]:
```suggestion
举个例子，如果使用 Vert.x 的 _launcher_ 类启动应用，则 classpath 应该设置为：
```

File: docs/translation/vertx-infinispan/java/index.adoc [hunk 3]
```diff
@@ -113,14 +116,14 @@ Vertx.clusteredVertx(options, res -> {
...
+    // 失败！
   }
 });
 ----
 
-In this case, vert.x is not the cache manager owner and so do not shut it down on close.
+在这种情况下，Vert.x 并不是缓存管理器的所有者，因此不能在关闭 Vert.x 时停止 Infinispan 。
 
-Notice that the custom Infinispan instance need to be configured with:
+需要注意的是，需要通过如下配置来自定义 Infinispan 实例：
```
Comment (chengenzhao) [inline docs/translation/vertx-infinispan/java/index.adoc]:
```suggestion
请注意，可通过如下配置以定制 Infinispan 实例：
```

File: docs/translation/vertx-infinispan/java/index.adoc [hunk 4]
```diff
@@ -289,93 +299,104 @@ element of the configuration. For example:
...
+设置正确的网络接口。
 
-=== When multicast is not available
+[[_when_multicast_is_not_available]]
+=== 组播不可用
 
-In some cases you may not be able to use multicast discovery as it might not be available in your environment.
-In that case you should configure another protocol, e.g. `TCPPING` to use TCP sockets, or `S3_PING` when running on Amazon EC2.
+在某些情况下，由于特殊的运行环境，可能无法使用组播。
+在这种情况下，应该配置为其他协议，例如配置 `TCPPING` 以使用 TCP 套接字，或配置 `S3_PING` 以使用亚马逊 S3。
```
Comment (chengenzhao) [inline docs/translation/vertx-infinispan/java/index.adoc]: 原文是ec2，是原文的错误吗？

File: docs/translation/vertx-infinispan/java/index.adoc [hunk 5]
```diff
@@ -385,66 +406,70 @@ ReadStream<V> valueStream = infinispanAsyncMap.valueStream();
...
+换句话说，它可以移动数据以适应新的集群拓扑。
+此过程可能需要一些时间，具体取决于集群数据量和节点数量。
 
-=== Split-brain syndrome
+=== 脑裂
 
-In a perfect world, there would be no network equipment failures.
-Reality is, though, that sooner or later your cluster will be divided into smaller groups, unable to see each others.
+在理想环境中，不会出现网络设备故障。
+实际上，集群早晚会被分成多个小组，彼此之间不可见。
```
Comment (chengenzhao) [inline docs/translation/vertx-infinispan/java/index.adoc]:
```suggestion
实际上，集群迟早会被分成多个小组，彼此之间不可见。
```

**Review by chengenzhao — APPROVED** [review]

File: docs/translation/vertx-infinispan/java/index.adoc [hunk 6]
```diff
@@ -385,66 +406,70 @@ ReadStream<V> valueStream = infinispanAsyncMap.valueStream();
...
 
-When a member joins or leaves the cluster, Infinispan rebalances cache entries on the new set of members.
-In other words, it moves data around to accomodate the new cluster topology.
-This process may take some time, depending on the amount of clustered data and number of nodes.
+当有节点加入或离开集群时，Infinispan 会在新的集群拓扑中重新平衡分配（rebalance）缓存条目。
+换句话说，它可以移动数据以适应新的集群拓扑。
+此过程可能需要一些时间，具体取决于集群数据量和节点数量。
 
-=== Split-brain syndrome
+=== 脑裂
```
Comment (okou19900722) [inline docs/translation/vertx-infinispan/java/index.adoc:426]:
```suggestion
[[_split_brain_syndrome]]
=== 脑裂
```

File: docs/translation/vertx-infinispan/java/index.adoc [hunk 7]
```diff
@@ -289,93 +299,104 @@ element of the configuration. For example:
...
+=== 组播不可用
 
-In some cases you may not be able to use multicast discovery as it might not be available in your environment.
-In that case you should configure another protocol, e.g. `TCPPING` to use TCP sockets, or `S3_PING` when running on Amazon EC2.
+在某些情况下，由于特殊的运行环境，可能无法使用组播。
+在这种情况下，应该配置为其他协议，例如配置 `TCPPING` 以使用 TCP 套接字，或配置 `S3_PING` 以使用亚马逊 S3。
 
-For more information on available JGroups discovery protocols and how to configure them please consult the http://www.jgroups.org/manual/index.html#Discovery[JGroups documentation].
+有关其他可用的 JGroups 发现协议及其如何配置的更多信息，
+请查阅 http://www.jgroups.org/manual/index.html#Discovery[JGroups文档] 。
```
Comment (okou19900722) [inline docs/translation/vertx-infinispan/java/index.adoc]:
```suggestion
有关其他可用的 JGroups 发现协议及其如何配置的更多信息，请查阅 http://www.jgroups.org/manual/index.html#Discovery[JGroups文档] 。
```

**Review by okou19900722 — CHANGES_REQUESTED** [review]

File: docs/translation/vertx-infinispan/java/index.adoc (hunk 4 above)
Comment (Leibnizhu) [reply docs/translation/vertx-infinispan/java/index.adoc]: 翻了下官方文档： https://infinispan.org/docs/stable/titles/configuring/configuring.html#preconfigured_jgroups_stacks-configuring ，原文应该没错，我翻译错了，改回去：

> 

default-jgroups-ec2.xml | ec2 | Uses TCP for transport and S3_PING for discovery. Suitable for Amazon EC2 nodes where UDP multicast is not available.
-- | -- | --

**Pull request merged**